### ISSUE TITLE
Issue to track creation of a Blazor WebConsole Driver and associated components

### ISSUE THREAD

**Issue by FreddieMercurial** [migueldeicaza/gui.cs#1642]
Per side conversation in #1601, creating this ticket to track development of this feature.

At the moment, it seems like the Blazor required references might be best kept in a separate project.
I'm thinking I may not necessarily need to fork.

I may create a new repo that is more closely aligned to Gui.cs and does not also accommodate Spectre.console (that'll be a fork I suppose)..

Primary repos (at the moment)
https://github.com/HACC-ers/HACC
https://github.com/HACC-ers/HACC.Development

**Comment by BDisp** [comment]
I made this changes in gui.cs:
https://github.com/BDisp/gui.cs/commit/4921492db6bfa49ad70d4a8ad4a7af118150c518

The changes in HACC:
https://github.com/BDisp/HACC/commit/8bfc71b110e1d1152aec4ea89a9898764fb84c39

The changes in HACC.Development:
https://github.com/BDisp/HACC.Development/commit/70d98466ef619841078f9f41036a6344223de77a

**Comment by tig** [comment]
Would it be better for this Issue to be renamed to "Add HTML5 rendering support"???  I think that's a more customer-centric way of stating it, no?

**Comment by FreddieMercurial** [comment]
Unfortunately it depends presently on a lot of newer DotNet 6 features, which could maybe be avoided- but having the demo on bleeding edge Blazor seems to make sense-ish?

**Comment by FreddieMercurial** [comment]
Some seven days later, BDisp and I have made a lot of progress on this and have it a little better distilled into a usable package.
We still have a ways to go on finishing the implementation but we're a heck of a lot closer!

**Comment by migueldeicaza** [comment]
My take is that something like HTML5 rendering support as HACC implements is a good candidate for an add-on to Gui.cs, in the same way that the terminal emulation is an add-on component to Gui.cs.

The rationale is that these are components with evolving APIs and evolving implementations, and should grow, mature and evolve on their own (also, they can be used with other libraries)

**Comment by FreddieMercurial** [comment]
Yes, it so happens that I was able to inherit your structure and implement Spectre.Console's Interface pattern at the same time. I think we're going to be a multi-compatible component. Your one stop blazor gui shop!

**Comment by FreddieMercurial** [comment]
PS @migueldeicaza  I am a big fan of yours. So sad you're not at Microsoft (for selfish reasons! Still had hoped to work with you).

But of course, I am not tied to microsoft... Maybe our paths will yet cross!

**Comment by FreddieMercurial** [comment]
We've basically formed a team/organization in the process of this and we're getting very close to a working version. Starting to display text in the right colors and places.

**Comment by FreddieMercurial** [comment]
https://github.com/Blazor-Console/HACC/blob/main/README.md
![image](https://user-images.githubusercontent.com/3766240/162673536-d71e2d55-6c2e-4adb-8073-723c8deb135a.png)

**Comment by tznind** [comment]
Looking great!

**Comment by Trolldemorted** [comment]
are the problems on the right border caused by non-monospaced fonts?

**Comment by BDisp** [comment]
> are the problems on the right border caused by non-monospaced fonts?

No, because it's a monospaced "Courier" font. It has to do with wide runes that not been yet well coded.

**Comment by tig** [comment]
Related to, but not the same as:

- #2938

**Comment by vonj** [comment]
Would this feature make it easier to implement another thing - the ability for a Terminal.GUI program to launch its own graphical window instead of having to rely on being launched inside a terminal or console?

**Comment by tig** [comment]
No.
